test/framework/modules.py

```diff
@@ -1229,6 +1287,22 @@ def test_module_use_unuse(self):
             self.assertEqual(os.environ['MODULEPATH'], test_dir1)
             self.modtool.unuse(test_dir1)
             self.assertFalse('MODULEPATH' in os.environ)
+            # Unuse when the MODULEPATH is already empty
+            self.modtool.unuse(test_dir1)
+            self.assertFalse('MODULEPATH' in os.environ)
+            os.environ['MODULEPATH'] = old_module_path  # Restore
+
+            # Forcing to use the module command reloads modules (Only Lmod does this)
+            old_module_path = os.environ['MODULEPATH']
```

Comment:
You are right, but those are more or less different (sub-)testcases. I wanted to make them self-contained, so it would still work if we remove one of those. Maybe best done via a fixture or context-manager but yeah...